xonsh/tools.py

```diff
@@ -205,6 +216,28 @@ def __eq__(self, other):
         return all(map(operator.eq, self, other))
 
 
+class EnvPathDir(EnvPath):
```

Comment:
Hey @rajathagasthya -- sorry for the radio silence. Yes, changing `EnvPath` to only include directories will cause `BASH_COMPLETIONS` to fail. My suggestion was to change the ensurers for `BASH_COMPLETIONS` to instead check that it's contents are either a directory or a file. 

In short, PATHS shouldn't contain files, generally, but `BASH_COMPLETIONS` is an exception. But we should special case the exception, not the general type. 

Does that make sense? (I'm not sure if it does, but it's in my head)